declaration/models.py

```diff
@@ -53,3 +54,16 @@ def __str__(self):
         return f'User: {self.user} - tracking code: {self.tracking_code}'
 
 
+class PackageStatusHistory(models.Model):
+    declaration = models.ForeignKey(PackageDeclaration, on_delete=models.CASCADE)
+    status = models.CharField(max_length=50)
```

Comment:
Wrong approach, you have a model of Status. Moreover, you need to create an entry automatically when the package changes its status.